05/app.js

```diff
@@ -0,0 +1,72 @@
+function Student (name, surname) {
+        this.name = name;
+        this.surname = surname;
+        this.grades = {};
+}
+
+Student.prototype.addSubject = function (subject) {
+        if (typeof this.grades[subject] === 'undefined')
+                return this.grades[subject] =[]
+}
+
+Student.prototype.addRating = function (num, subject) {
+        
+        if (typeof this.grades[subject] === 'undefined') {
+        console.log(`najpierw skorzystaj z funkcji addSubject aby dodać przedmiot`)
+        } else {
+                this.grades[subject].push(num);
+        }
+}
+        
+
+
+Student.prototype.averageOfSubjects = function (subject) {
+        let result = 0;
+        let length;
+        let x = 0;
+        subject = this.grades[subject];
+        if (subject) {
+                length = subject.length
+                subject.map(item => {
```

Comment:
Skoro operujemy na elementach tablicy to używamy `forEach`, `map` wykorzystujemy przy tworzeniu nowej tablicy